### PR TITLE
chore: update GET raw allocation to account for loss of workload information [DET-5973]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change updates GET raw allocation to account for the fact that "workloads" are no longer really stored, and we don't know the start times at which we began to produce some set of metrics.

## Test Plan
- [x] run some experiments, test the API
- [ ] ~~probably just deprecate the current behavior of this API before I land, so we can change it later when it is convenient (I imagine eventually this logic won't be right at all).~~

## Commentary (optional)
Follow up to make this all worthwhile is to remove the weird metric start time derivation from push.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by stoksc** [comment]
blocked on #2910

**Review by mackrorysd — APPROVED** [review]
Approve, but I think being explicit about that assumption will potentially save some trouble later.

File: master/static/srv/get_raw_allocation.sql [hunk 1]
```diff
@@ -22,25 +22,26 @@ workloads AS (
...
-                'checkpoint' AS kind,
-                trial_id,
-                tstzrange(start_time, end_time) AS range
-            FROM
-                raw_checkpoints
+            FROM (
+                     SELECT
+                         kind,
+                         trial_id,
+                         LAG(end_time, 1) OVER (PARTITION BY trial_id ORDER BY end_time) AS start_time,
```
[the remark this answers is not in the document]
Comment (stoksc) [reply master/static/srv/get_raw_allocation.sql:32]: cool, i'll drop a comment

File: master/static/srv/get_raw_allocation.sql [hunk 2]
```diff
@@ -22,25 +22,70 @@ workloads AS (
...
+                                'checkpointing' AS kind,
+                                trial_id,
+                                end_time
+                            FROM
+                                raw_checkpoints
+                        ) metric_reports
+                ) derived_workload_spans
+            WHERE
+                start_time IS NOT NULL
+                AND end_time IS NOT NULL
```
Comment (stoksc) [inline master/static/srv/get_raw_allocation.sql:87]: ended up noticing i had a bug where steps with weird failures that never got written got counted by any run of the query and blew up the results, so i just excluded them.

**Comment by stoksc** [comment]
Going to land this for now. Since the new version wouldn't work for pre-0.17.0 experiments at all, I'm going to land this and follow up with a `v2` or may just a newly named API endpoint (probably just `/api/v1/resources/allocations`) to supersede this.

**Pull request merged**